docs/data/tree-view/rich-tree-view/lazy-loading/lazy-loading.md

```diff
@@ -0,0 +1,36 @@
+---
+productId: x-tree-view
+title: Rich Tree View - Lazy Loading Children
+components: RichTreeView, TreeItem
+packageName: '@mui/x-tree-view'
+githubLabel: 'component: tree view'
+waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/treeview/
+---
+
+# Rich Tree View - Lazy Loading Children
+
+<p class="description">Lazy load the data from your Tree View.</p>
+
+## Basic usage
+
+To dynamically load data from the server, including lazy-loading of children, you must create a data source and pass the treeViewDataSource prop to the Rich Tree View.
+
+The data source also requires the `getChildrenCount()` attribute to handle tree data:
+
+getChildrenCount(): Returns the number of children for the item. If the children count is not available for some reason, but there are some children, returns -1.
+
+{{"demo": "BasicLazyLoading.js"}}
+
+## Data caching
+
+### Custom cache
+
+### Customize the cache lifetime
+
+The `TreeViewDataSourceCacheDefault` has a default Time To Live (`ttl`) of 5 minutes. To customize it, pass the ttl option in milliseconds to the `TreeViewDataSourceCacheDefault` constructor, and then pass it as the `treeViewDataSourceCache` prop.
```

Comment:
Not sure we should have caching by default, for the same reason as

Reply:
For the same reason as ? :angel: 

I guess we should follow the same default behavior as the grid here (I did not check there default behavior)

Reply:
Ah oops, I changed my mind on this, I forgot to delete this comment. It was initially about https://nextjs.org/blog/next-15#caching-semantics.